plugins/inputs/opensearch_query/opensearch_query.go

```diff
@@ -0,0 +1,197 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package opensearch_query
+
+import (
+	"context"
+	"crypto/tls"
+	_ "embed"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	influxtls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/opensearch-project/opensearch-go/v2"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// OpensearchQuery struct
+type OpensearchQuery struct {
+	URLs                []string        `toml:"urls"`
+	Username            string          `toml:"username"`
+	Password            string          `toml:"password"`
+	EnableSniffer       bool            `toml:"enable_sniffer"`
+	Timeout             config.Duration `toml:"timeout"`
+	HealthCheckInterval config.Duration `toml:"health_check_interval"`
+	Aggregations        []osAggregation `toml:"aggregation"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	influxtls.ClientConfig
+	osClient *opensearch.Client
+}
+
+// osAggregation struct
+type osAggregation struct {
+	Index                string          `toml:"index"`
+	MeasurementName      string          `toml:"measurement_name"`
+	DateField            string          `toml:"date_field"`
+	DateFieldFormat      string          `toml:"date_field_custom_format"`
+	QueryPeriod          config.Duration `toml:"query_period"`
+	FilterQuery          string          `toml:"filter_query"`
+	MetricFields         []string        `toml:"metric_fields"`
+	MetricFunction       string          `toml:"metric_function"`
+	Tags                 []string        `toml:"tags"`
+	IncludeMissingTag    bool            `toml:"include_missing_tag"`
+	MissingTagValue      string          `toml:"missing_tag_value"`
+	mapMetricFields      map[string]string
+	aggregationQueryList []aggregationQueryData
+}
+
+func (*OpensearchQuery) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init the plugin.
+func (o *OpensearchQuery) Init() error {
+	if o.URLs == nil {
+		return fmt.Errorf("opensearch urls is not defined")
+	}
+
+	err := o.connectToOpensearch()
+	if err != nil {
+		return fmt.Errorf("error connecting to opensearch: %w", err)
+	}
```

Comment:
Should this already be done while starting telegraf? I see this is also done in `Gather`.

I would rather change this to log a warning or remove this at all from `Init`.